### DISCUSSION TITLE
Add Support for Clear Linux OS

### DISCUSSION THREAD

**Discussion by antonym** [antonym/netboot.xyz#500] [Ideas]
Add support for Clear Linux OS:

https://clearlinux.org/

**Comment by thelamer** [comment]
pxe tarballs https://cdn.download.clearlinux.org/releases/32020/clear/clear-32020-pxe.tar.xz

**Comment by thelamer** [comment]
This pxe booter is 100% automated with a yaml and is always data destructive. 
https://github.com/clearlinux/clr-installer/blob/master/scripts/installer.yaml
From an internet based perspective people would need to provide `YAML_URL` below to even boot this properly. 

Boot snippet: 
```
kernel linux quiet init=/usr/lib/systemd/systemd-bootchart initcall_debug tsc=reliable no_timer_check noreplace-smp rw clri.descriptor=${YAML_URL} initrd=initrd
initrd initrd
```

I can host the assets , but wanted to make sure you want to add this to the menus given this info.

**Comment by antonym** [comment]
So we'd probably need to add an option for overriding the default yaml and possibly make a note saying it's destructive?  Are there any live options that don't modify the disk?

**Comment by thelamer** [comment]
No we can't provide the yaml even it is completely custom.
They need to make the yaml and host it somewhere

**Comment by antonym** [comment]
That's what I mean, just an option to specify your custom yaml location if the user wants to override the default.

**Comment by thelamer** [comment]
Their is no default it is non functional without a custom yaml

**Comment by antonym** [comment]
Ah, I thought https://github.com/clearlinux/clr-installer/blob/master/scripts/installer.yaml was the default... well we can table this for now if we want then.

**Comment by antonym** [comment]
Maybe we just switch to the ISO installer:

https://docs.01.org/clearlinux/latest/get-started/bare-metal-install-server.html

**Comment by thelamer** [comment]
There is no way to retrieve a squash with their hooks in that ISO. (they do their own custom init scripts and hooks nothing for ingesting a squash from a remote server) 
That pxe tarball has a fat initrd with a bunch of shit in it, but the logic all revolves around loading an automated installer yaml for the variables to use for the non interactive installation. 

See here : https://github.com/clearlinux/distribution/issues/1371

**Comment by dragon788** [comment]
According to their PXE installation documentation, it is possible to boot without automatically installing, but that documentation could be out of date. 

https://docs.01.org/clearlinux/latest/get-started/ipxe-install.html#add-clr-installer-yaml-configuration-files
In the Note immediately above this section heading it mentions that without a directive it will just boot.
